src/main/java/org/openstreetmap/atlas/utilities/caching/ResourceCache.java

```diff
@@ -0,0 +1,208 @@
+package org.openstreetmap.atlas.utilities.caching;
+
+import java.net.URI;
+import java.util.Optional;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.streaming.resource.Resource;
+import org.openstreetmap.atlas.utilities.caching.strategies.CachingStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The basic resource cache implementation. This cache loads a resource using a given {@link URI},
+ * {@link CachingStrategy}, and default {@link ResourceFetchFunction}. Since using raw URIs can
+ * often be cumbersome, users of this class are encouraged to extend it and encapsulate URI
+ * construction in appropriate subclass helpers.
+ *
+ * @author lcram
+ */
+public class ResourceCache
+{
+    private static final Logger logger = LoggerFactory.getLogger(ResourceCache.class);
+
+    private URI resourceURI;
+    private CachingStrategy cachingStrategy;
+    private ResourceFetchFunction defaultFetcher;
```

Comment:
type of ```defaultFetcher``` can be ```Function<URI, Resource>``` instead of defining your own type. What do you think about it?

Reply:
Good idea. @matthieun suggested this as well.